lib/tasks/TaskScheduler.js

```diff
@@ -22,18 +31,28 @@ class TaskScheduler {
      * @param {function} [getDedupeKeyFunc] - function to get the dedupe
      *   key of a task: getDedupeKeyFunc(ctx) -> {string} key
      *   (see: {@link TaskScheduler.push()})
+     * @param {number} [noQueueKeyConcurrency] - concurrency for tasks
+     *   without a queue key, default has no limit
      */
-    constructor(processingFunc, getQueueKeyFunc, getDedupeKeyFunc) {
+    constructor(processingFunc, getQueueKeyFunc, getDedupeKeyFunc, noQueueKeyConcurrency) {
         this._processingFunc = processingFunc;
         this._getQueueKeyFunc = getQueueKeyFunc;
         this._getDedupeKeyFunc = getDedupeKeyFunc;
+        this._drainFunc = defaultDrain;
         this._taskQueues = {};
         this._dedupeCache = {};
+        if (noQueueKeyConcurrency) {
+            this._noQueueKeyTasks = async.queue(this._processingFunc, noQueueKeyConcurrency);
+            this._noQueueKeyTasks.drain = this._tryDrain.bind(this);
+        }
```

Comment:
what is the point?
it should not be the responsability of the TaskScheduler to have multiple implementations :
- either it degrades seamlessly (but we go through the same code), e.g. if getDedupeKeyFunc() returns something unique or maxQueueSize == 1;
- or we really need to have multiple implementations, and we should have an abstraction and multiple implementations (and the proper one is created on BackbeatConsumer startup)